docs/data-sources/kafka_cluster_data.md

```diff
@@ -0,0 +1,50 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "upstash_kafka_cluster_data Data Source - terraform-provider-upstash"
+subcategory: ""
+description: |-
+  
+---
+
+# upstash_kafka_cluster_data (Data Source)
+
+
+
+## Example Usage
+
+```terraform
+data "upstash_kafka_cluster_data" "clusterData" {
+  cluster_id = resource.upstash_kafka_cluster.exampleCluster.cluster_id
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- **cluster_id** (String) Unique Cluster ID for requested cluster
+
+### Optional
+
+- **id** (String) The ID of this resource.
+
+### Read-Only
+
+- **cluster_name** (String) Name of the cluster
+- **encoded_username** (String) Encoded Username for the cluster
+- **max_message_size** (Number) Max Message Size for the cluster
+- **max_messages_per_second** (Number) Max Messages Per Second for the cluster
+- **max_partitions** (Number) Max Partitions for the cluster
+- **max_retention_size** (Number) Max Retention Size of the cluster
+- **max_retention_time** (Number) Max Retention Time of the cluster
+- **multizone** (Boolean) Whether cluster has multizone attribute
```

Comment:
It might be better to say something like whether multizone replication enabled or not